### PR TITLE
Fix for task execution- and task roles.

### PR DESCRIPTION
**Pull request by glennbech**
* Task role was omitted from the service, causing applications using SDK's or APIs within containers to be unable to get credentials 
* service execution role (needed by ECS) and Task role (needed by application)  is now clearly separated 
* It seems that Service role is not used, I think we can remove it as well. See comments in terraform file main.tf

### CODE REVIEW

File: ecs/fargate/main.tf [hunk 1]
```diff
@@ -13,7 +13,7 @@ resource "aws_cloudwatch_log_group" "main" {
 }
 
 # ------------------------------------------------------------------------------
-# IAM
+# IAM - Service role- is this used?
```
Comment (mikael-lindstrom) [inline ecs/fargate/main.tf]: It does not seem to be used here but it should, we are using it in the regular service module: https://github.com/TeliaSoneraNorge/telia-terraform-modules/blob/master/ecs/service/main.tf#L38
Could you add it here or should we do it in another PR?

File: ecs/fargate/main.tf [hunk 2]
```diff
@@ -26,6 +26,42 @@ resource "aws_iam_role_policy" "service_permissions" {
...
+# ------------------------------------------------------------------------------
+resource "aws_iam_role" "execution" {
+  name               = "${var.prefix}-task-execution-role"
+  assume_role_policy = "${data.aws_iam_policy_document.task_assume.json}"
+}
+
+resource "aws_iam_role_policy" "task_execution" {
+  name   = "${var.prefix}-task-execution"
+  role   = "${aws_iam_role.execution.id}"
+  policy = <<EOF
```
Comment (mikael-lindstrom) [inline ecs/fargate/main.tf]: Should we put this policy as a `data "aws_iam_policy_document"` in policies.tf to be consistent?

**Comment by glennbech** [comment]
I totally agree. I'll add the ```data "aws_iam_policy_document"```and the iam_role in this PR. I'll give you a heads up when done.

**Comment by glennbech** [comment]
@mikael-lindstrom regarding iam_role for service; (https://www.terraform.io/docs/providers/aws/r/ecs_service.html)  

> This parameter is required if you are using a load balancer with your service, but only if your task definition does not use the awsvpc network mode. If using awsvpc network mode, do not specify this role.

We don't need the role since we are using awsvpc networking mode for fargate, if I understand the details correctly.

File: ecs/fargate/main.tf (hunk 1 above)
Comment (glennbech) [reply ecs/fargate/main.tf]: @mikael-lindstrom regarding iam_role for service; (https://www.terraform.io/docs/providers/aws/r/ecs_service.html)

This parameter is required if you are using a load balancer with your service, but only if your task definition does not use the awsvpc network mode. If using awsvpc network mode, do not specify this role.

We don't need the role since we are using awsvpc networking mode for fargate, if I understand the details correctly.

**Comment by glennbech** [comment]
@mikael-lindstrom I think this is ready now.

File: ecs/fargate/main.tf (hunk 1 above)
Comment (mikael-lindstrom) [reply ecs/fargate/main.tf]: You are correct 🥇 We can remove this, either we do it here or in another PR. Approving so you can merge if we should do this in another PR.

**Comment by glennbech** [comment]
Remved serivce permissions as well and merged.
